component/src/main/java/com/celements/structEditor/fields/DateTimePageType.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.celements.structEditor.fields;
+
+import static com.celements.structEditor.classes.DateTimePickerEditorClass.*;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.cells.attribute.AttributeBuilder;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.celements.model.object.xwiki.XWikiObjectFetcher;
+import com.celements.structEditor.classes.DateTimePickerEditorClass.Type;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+@Component(DateTimePageType.PAGETYPE_NAME)
+public class DateTimePageType extends AbstractStructFieldPageType {
+
+  public static final String PAGETYPE_NAME = "DateTime";
+
+  static final String VIEW_TEMPLATE_NAME = "DateTimeView";
+
+  @Override
+  public String getName() {
+    return PAGETYPE_NAME;
+  }
+
+  @Override
+  protected String getViewTemplateName() {
+    return VIEW_TEMPLATE_NAME;
+  }
+
+  @Override
+  public Optional<String> tagName() {
+    return Optional.of("cel-input-date-time");
+  }
+
+  private static final Map<Type, String> PICKER_TYPE_CSS_CLASS_MAP = ImmutableMap.of(
+      Type.DATE_PICKER, "cel_datePicker",
+      Type.TIME_PICKER, "cel_timePicker",
+      Type.DATE_TIME_PICKER, "cel_dateTimePicker");
+
+  @Override
+  public void collectAttributes(AttributeBuilder attrBuilder, DocumentReference cellDocRef) {
+    attrBuilder.addNonEmptyAttribute("type", "text");
+    try {
+      Optional<Date> cellValue = getStructDataEditorService().getCellDateValue(cellDocRef,
+          modelContext.getCurrentDoc().orNull());
+      if (cellValue.isPresent()) {
+        Optional<String> dateFormat = getStructDataEditorService().getDateFormatFromField(
+            cellDocRef);
+        String value = null;
+        if (dateFormat.isPresent()) {
+          value = new SimpleDateFormat(dateFormat.get()).format(cellValue.get());
```

Comment:
use `com.celements.common.date.DateFormat` instead, it internally caches Formatter instances.